deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -123,7 +123,7 @@ def fit(self,
           max_checkpoints_to_keep=5,
           checkpoint_interval=1000):
     return self.fit_generator(
-        self.default_generator(dataset, epochs=nb_epoch),
+        self.default_generator(dataset, epochs=nb_epoch, deterministic=False),
```

Comment:
Might be good to make deterministic a kwarg set to True. That way, users can turn off if desired